src/libcore/fmt/mod.rs

```diff
@@ -180,8 +180,9 @@ enum Void {}
            issue = "0")]
 #[doc(hidden)]
 pub struct ArgumentV1<'a> {
-    value: &'a Void,
-    formatter: fn(&Void, &mut Formatter) -> Result,
+    _ph: PhantomData<&'a ()>,
```

Comment:
I feel like a better fix would be to use `&'a ()`. And perhaps `Void` ought to be defined not as an empty enum, but rather as a zero-sized struct that exposes no constructor? This is sort of an interesting question. I'm actually not sure what's the best way to model the equivalent of `void*`. Seems like something we can discuss when it comes to the unsafe code guidelines. =)